src/main/java/net/dv8tion/jda/internal/requests/restaction/InviteActionImpl.java

```diff
@@ -75,6 +83,12 @@ public InviteActionImpl setMaxAge(final Integer maxAge)
         if (maxAge != null)
             Checks.notNegative(maxAge, "maxAge");
 
+        if (nonCommunityRestricted)
+        {
+            Checks.check(maxAge != null && maxAge >= 1 && maxAge <= MAX_RESTRICTED_AGE,
+                "maxAge for invites in non-community guilds must be between 1 second and 30 days!");
```

Comment:
```suggestion
                "The maximum age for invites in non-community guilds must be between 1 second and 30 days!");
```

Also, the [API Docs](https://discord.com/developers/docs/resources/channel#create-channel-invite) say 7 days is the maximum, are the docs wrong?

Reply:
Seems like they haven't merged the API doc changes yet? Not sure if the changes are live

Reply:
The client also limits it to 7 days:

![image](https://github.com/discord-jda/JDA/assets/18090140/ef6b3be2-22d4-4e25-82bb-65fb6ec00748)

Reply:
https://github.com/discord/discord-api-docs/pull/6233/files#diff-160acb4654c611c67c4ee46b2ce5dda2be82212af8d1d00cd0256ef6edda1e95L1177

This line changes the docs value from 7 days to 30 days, but hasn't been merged yet